website/addons/owncloud/static/owncloudNodeConfig.js

```diff
@@ -0,0 +1,472 @@
+var ko = require('knockout');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var $osf = require('js/osfHelpers');
+
+var $modal = $('#ownCloudCredentialsModal');
+var oop = require('js/oop');
+var FolderPickerViewModel = require('js/folderPickerNodeConfig');
+
+var ViewModel = oop.extend(FolderPickerViewModel,{
```

Comment:
Consider `oop.extend`ing `OauthAddonFolderPicker`. This will dramatically reduce the amount of code needed in this file. Recent changes on develop make [`s3NodeConfig.js`](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/addons/s3/static/s3NodeConfig.js#L15) a good example of how to extend that viewmodel to support addons that don't support OAuth.